fenix/app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -1051,11 +1064,16 @@ class HomeFragment : Fragment() {
     }
 
     private fun observeWallpaperUpdates() {
-        consumeFrom(requireComponents.appStore) {
-            val currentWallpaper = it.wallpaperState.currentWallpaper
-            if (currentWallpaper.name != lastAppliedWallpaperName) {
-                applyWallpaper(wallpaperName = currentWallpaper.name, orientationChange = false)
-            }
+        consumeFlow(requireComponents.appStore, viewLifecycleOwner) { flow ->
```

Comment:
Updated this to make the flow more lifecycle aware, to stop when the lifecycler owner is destroyed based on consumeFlow's implementaiton.